File: puns-back-parent/puns-back-core/src/main/java/com/lukascode/puns/core/api/dto/AddScoreRequest.java
package com.lukascode.puns.core.api.dto;

import javax.validation.constraints.NotNull;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AddScoreRequest {

    @NotNull
    private Long playerId;

    @NotNull
    private Integer points;

    private String gameRoomId;

    private String guessedWord;

}

File: puns-live-server/src/boundary/add-score-request.js
module.exports = class AddScoreRequest {

    constructor(playerId, points, gameRoomId, guessedWord) {
        this.playerId = playerId;
        this.points = points;
        this.gameRoomId = gameRoomId;
        this.guessedWord = guessedWord;
    }

}